spec/system/page_objects/pages/user.rb

```diff
@@ -24,11 +24,32 @@ def has_warning_messages_path?(user)
         page.has_current_path?("/u/#{user.username}/messages/warnings")
       end
 
+      def staff_info_section
+        begin
+          page.find(".staff-counters")
+        rescue Capybara::ElementNotFound
+          nil
+        end
+      end
+
       def click_staff_info_warnings_link(user, warnings_count: 0)
-        staff_counters = page.find(".staff-counters")
-        staff_counters.find("a[href='/u/#{user.username}/messages/warnings']").click
+        staff_info_section.find("a[href='/u/#{user.username}/messages/warnings']").click
```

Comment:
I'm curious, would `.find(".staff-counters a[href='/u/#{user.username}/messages/warnings']")` also work? Are there downsides/upsides to doing one `find()` vs two?

Reply:
Yes, having a single `find` also works for the test case which is using `click_staff_info_warnings_link`, as does the previous version with two `find`s.

 Doing a single `find` works well enough for scenarios where we don't test for or don't expect the first element to be missing from the page (otherwise it throws a `Capybara::ElementNotFound`), and would be a more concise syntax than the version with two `find`s. 

For the original version with two `find`s, one benefit is being able to instantly tell where the missing element is from - for example: with a single `find`, the error message would be something like `Capybara::ElementNotFound: Unable to find css ".staff-counters <anchor selector here>"` as opposed to `Capybara::ElementNotFound: Unable to find css "<EITHER .staff-counters OR <anchor selector>>`.

The other reason we might want to have two `find`s is to make it clearer that we are first finding the `.staff-counters` element (and I chose to house it under a separate method `staff_info_section` for reuse it in `staff_info_flagged_posts_counter`) before looking for another element within that section. 

aside: separating into a different method opens up the option to better handle cases where that `.staff-counters` element is not expected to be present and return a value (or null object) to be tested (though this could also be handled in the test file with the `raise_error` matcher). 

--
Overall, I think a single `find` would trade off being able to keep the page object more concise against some complexity/more verbose handling in the test file.